src/test/java/io/strimzi/kafka/bridge/http/ConsumerSubscriptionIT.java

```diff
@@ -355,6 +442,105 @@ void assignAfterSubscriptionTest(VertxTestContext context) throws InterruptedExc
+        JsonArray partitions = new JsonArray();
+        JsonObject part0 = new JsonObject();
+        part0.put("topic", topic);
+        part0.put("partition", 0);
+
+        JsonObject part1 = new JsonObject();
+        part1.put("topic", topic);
+        part1.put("partition", 1);
+        partitions.add(part0);
+        partitions.add(part1);
+
+        partitionsRoot.put("partitions", partitions);
+
+        consumerService()
+                .createConsumer(context, groupId, json);
+
+        CompletableFuture<Boolean> assignCF = new CompletableFuture<>();
+        consumerService()
+                .assignRequest(groupId, name, partitionsRoot)
+                .sendJsonObject(partitionsRoot, ar -> {
+                    context.verify(() -> {
+                        assertThat(ar.succeeded(), is(true));
+                        HttpResponse<JsonObject> response = ar.result();
+                        assertThat(response.statusCode(), is(HttpResponseStatus.NO_CONTENT.code()));
+                    });
+                    assignCF.complete(true);
+                });
+
+        assignCF.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        // Validate the existing consumer list
+        CompletableFuture<Boolean> listSubscriptionsBefore = new CompletableFuture<>();
+        consumerService()
+                .listSubscriptionsConsumerRequest(groupId, name)
+                .as(BodyCodec.jsonObject())
+                .send(ar -> {
+                    context.verify(() -> {
+                        assertThat(ar.succeeded(), is(true));
+                        HttpResponse<JsonObject> response = ar.result();
+                        assertThat(response.statusCode(), is(HttpResponseStatus.OK.code()));
+                        assertThat(response.body().getJsonArray("topics").size(), is(1));
+                    });
+                    listSubscriptionsBefore.complete(true);
+                });
+        listSubscriptionsBefore.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        JsonObject emptyPartitionsRoot = new JsonObject();
+        JsonArray emptyPartitions = new JsonArray();
+        emptyPartitionsRoot.put("partitions", emptyPartitions);
+
+        CompletableFuture<Boolean> assignEmptyCF = new CompletableFuture<>();
+        consumerService()
+                .assignRequest(groupId, name, emptyPartitionsRoot)
+                .sendJsonObject(emptyPartitionsRoot, ar -> {
+                    context.verify(() -> {
+                        assertThat(ar.succeeded(), is(true));
+                        HttpResponse<JsonObject> response = ar.result();
+                        assertThat(response.statusCode(), is(HttpResponseStatus.NO_CONTENT.code()));
+                    });
+                    assignEmptyCF.complete(true);
+                });
+
+        assignEmptyCF.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        // Validate the existing consumer list
+        CompletableFuture<Boolean> listSubscriptionsAfter = new CompletableFuture<>();
```

Comment:
same as before you should be able to use just one `listSubscriptions` completable future for before and after, just creating a new instance assigning to the same variable.

Reply:
Hi @ppatierno , that does not work here. it shows "Local variable listSubscriptionsBefore defined in an enclosing scope must be final or effectively finalJava(536871575)"